### PR TITLE
Refactoring

### PR DESCRIPTION
**Pull request by outscale-toa**

### CODE REVIEW

File: outscale/data_source_outscale_vm.go [hunk 1]
```diff
@@ -351,28 +347,24 @@ func getOApiVMAttributesSchema() map[string]*schema.Schema {
...
 		},
 		"keypair_name": {
 			Type:     schema.TypeString,
-			Optional: true,
 			Computed: true,
 		},
 		"security_group_ids": {
 			Type:     schema.TypeSet,
-			Optional: true,
+			Computed: true,
```
Comment (outscale-mgo) [inline outscale/data_source_outscale_vm.go:362]: is it okay, to change that from optional to computed ?

File: outscale/data_source_outscale_vm.go [hunk 2]
```diff
@@ -351,28 +347,24 @@ func getOApiVMAttributesSchema() map[string]*schema.Schema {
...
 			Elem:     &schema.Schema{Type: schema.TypeString},
 		},
 		"security_group_names": {
 			Type:     schema.TypeList,
-			Optional: true,
```
Comment (outscale-mgo) [inline outscale/data_source_outscale_vm.go:375]: same here ?

File: outscale/data_source_outscale_vm.go [hunk 3]
```diff
@@ -860,7 +833,7 @@ func getOApiVMAttributesSchema() map[string]*schema.Schema {
 		},
 		"private_ips": {
 			Type:     schema.TypeList,
-			Optional: true,
+			Computed: true,
```
Comment (outscale-mgo) [inline outscale/data_source_outscale_vm.go:836]: same here ?

File: outscale/resource_outscale_nic.go [hunk 4]
```diff
@@ -607,6 +596,9 @@ func resourceOutscaleOAPINicUpdate(d *schema.ResourceData, meta interface{}) err
 		pips := d.Get("pips").(*schema.Set).List()
 		prips := pips[:0]
 		pip := d.Get("private_ip")
+		input := oscgo.LinkPrivateIpsRequest{
+			NicId: d.Id(),
```
Comment (outscale-mgo) [inline outscale/resource_outscale_nic.go:600]: doesn't this change the result if `diff < 0`

File: outscale/data_source_outscale_vm.go (hunk 1 above)
Comment (outscale-toa) [reply outscale/data_source_outscale_vm.go:362]: Yes, that is the data_source, we cannot set  parameters directly.

File: outscale/data_source_outscale_vm.go (hunk 3 above)
Comment (outscale-toa) [reply outscale/data_source_outscale_vm.go:836]: Here too

File: outscale/data_source_outscale_vm.go (hunk 2 above)
Comment (outscale-toa) [reply outscale/data_source_outscale_vm.go:375]: Here too

**Review by outscale-mgo — APPROVED** [review]

**Pull request merged**